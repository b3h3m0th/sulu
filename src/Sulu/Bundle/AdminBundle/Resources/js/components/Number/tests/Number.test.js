// @flow
import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Number from '../Number';

test('Number should render', () => {
    const {container} = render(<Number onChange={jest.fn()} value={undefined} />);
    expect(container).toMatchSnapshot();
});

test('Number should render when disabled', () => {
    const {container} = render(<Number disabled={true} onChange={jest.fn()} value={8} />);
    expect(container).toMatchSnapshot();
});

test('Number should call onChange with parsed value', () => {
    const onChange = jest.fn();
    render(<Number onChange={onChange} value={2} />);

    const input = screen.queryByDisplayValue(2);
    fireEvent.keyDown(input, {target: {value: '10.2'}});

    expect(input).toHaveValue(10.2);
});

test('Number should call onChange with undefined when value isn`t a float', () => {
    const onChange = jest.fn();
    render(<Number onChange={onChange} value={2} />);

    const input = screen.queryByDisplayValue(2);
    fireEvent.change(input, {target: {value: 'text'}});

    expect(onChange).toHaveBeenCalledWith(undefined, expect.anything());
});

test('Number should call onChange with undefined when value is undefined', () => {
    const onChange = jest.fn();
    render(<Number onChange={onChange} value={2} />);

    const input = screen.queryByDisplayValue(2);
    fireEvent.change(input, {target: {value: undefined}});

    expect(onChange).toBeCalledWith(undefined, expect.anything());
});
