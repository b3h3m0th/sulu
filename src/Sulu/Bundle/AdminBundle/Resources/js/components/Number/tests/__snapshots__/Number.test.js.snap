// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number should render 1`] = `
<div>
  <label
    class="input default left"
  >
    <input
      type="number"
      value=""
    />
  </label>
</div>
`;

exports[`Number should render when disabled 1`] = `
<div>
  <label
    class="input default left disabled"
  >
    <input
      disabled=""
      type="number"
      value="8"
    />
  </label>
</div>
`;
